src/realm/table.cpp

```diff
@@ -2615,9 +2618,6 @@ void Table::clear()
 
         remove_backlink_broken_rows(state); // Throws
     }
-
```

Comment:
I believe the unit tests are failing because the Replication instruction must be called last in this method. When backlinks are involved in a clear instruction, the linked column must have the value cleared first with the origin table fully intact so that the linked rows are essentially erased first and then clear is applied to the remaining unlinked rows. This probably means you will have to store the original number of rows at the top of the method and then pass it in like `real->clear_table(this, previous_num_rows);`